### PR TITLE
add context support

### PR DESCRIPTION
**Pull request by JaSei**
based on #33

### CODE REVIEW

File: retry_test.go [hunk 1]
```diff
@@ -260,3 +260,16 @@ func TestCombineDelay(t *testing.T) {
...
+	start := time.Now()
+	ctx, cancel := context.WithCancel(context.Background())
+	cancel()
+	err := Do(
+		func() error { return errors.New("test") },
+		Context(ctx),
+	)
+	dur := time.Since(start)
+	assert.Error(t, err)
+	assert.True(t, dur < time.Millisecond, "immediately cancellation")
```
Comment (sranka) [inline retry_test.go:274]: I guess that anything less than the DefaultDelay might better express the expectation. We might also test that the function is called at most once.

**Review by sranka — APPROVED** [review]
LGTM

File: retry_test.go (hunk 1 above)
Comment (JaSei) [reply retry_test.go:274]: Thanks to test hints. This implementation doesn't work well. Retry is called many times and return an error. I think about what would return this code? error? or nothing?

Comment (JaSei) [reply retry_test.go]: I fixed it now - I think...

**Review by sranka — APPROVED** [review]

**Pull request merged**